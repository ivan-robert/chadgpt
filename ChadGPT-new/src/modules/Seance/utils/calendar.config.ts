import { LocaleConfig } from "react-native-calendars";

export const configCalendar = () => {
  LocaleConfig.locales["fr"] = {
    monthNames: [
      "Janvier",
      "Février",
      "Mars",
      "Avril",
      "Mai",
      "Juin",
      "Juillet",
      "Août",
      "Septembre",
      "Octobre",
      "Novembre",
      "Décembre",
    ],
    monthNamesShort: [
      "Janv.",
      "Févr.",
      "Mars",
      "Avril",
      "Mai",
      "Juin",
      "Juil.",
      "Août",
      "Sept.",
      "Oct.",
      "Nov.",
      "Déc.",
    ],
    dayNames: [
      "Dimanche",
      "Lundi",
      "Mardi",
      "Mercredi",
      "Jeudi",
      "Vendredi",
      "Samedi",
    ],
    dayNamesShort: ["Dim.", "Lun.", "Mar.", "Mer.", "Jeu.", "Ven.", "Sam."],
    today: "Aujourd'hui",
  };

  LocaleConfig.locales["en"] = {
    monthNames: [
      "January",
      "February",
      "March",
      "April",
      "May",
      "June",
      "July",
      "August",
      "September",
      "October",
      "November",
      "December",
    ],
    monthNamesShort: [
      "Jan.",
      "Feb.",
      "Mar.",
      "Apr.",
      "May",
      "Jun.",
      "Jul.",
      "Aug.",
      "Sep.",
      "Oct.",
      "Nov.",
      "Dec.",
    ],
    dayNames: [
      "Sunday",
      "Monday",
      "Tuesday",
      "Wednesday",
      "Thursday",
      "Friday",
      "Saturday",
    ],
    dayNamesShort: ["Sun.", "Mon.", "Tue.", "Wed.", "Thu.", "Fri.", "Sat."],
    today: "Today",
  };
};
