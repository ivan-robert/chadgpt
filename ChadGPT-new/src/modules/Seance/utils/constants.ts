export const SAVE_SEANCE_KEY = "SAVE SEANCE";

export const RETRIEVE_SEANCE_KEY = "RETRIEVE_SEANCE";
