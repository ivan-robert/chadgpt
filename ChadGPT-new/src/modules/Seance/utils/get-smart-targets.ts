import { Workout } from "#modules/Seance/domain/seance.types";
import { Exo } from "#shared/exo/domain/exo.types";
import { BodyPartDTO } from "#shared/exo/infra/muscles.dto";

export const getSmartTargets = (workout: Workout): BodyPartDTO[] => {
  const workoutExercises = getExercises(workout);

  const acc: Partial<Record<BodyPartDTO, number>> = {};
  workoutExercises.forEach((exo) => {
    if (exo.bodyPart in acc) {
      acc[exo.bodyPart]! += 1;
    } else {
      acc[exo.bodyPart] = 1;
    }
  });

  return Object.entries(acc)
    .map(([bodyPart, count]) => ({
      bodyPart: bodyPart as BodyPartDTO,
      count,
    }))
    .sort((a, b) => b.count - a.count)
    .slice(0, 2)
    .map((item) => item.bodyPart);
};

const getExercises = (workout: Workout): Exo[] => {
  const out: Exo[] = [];
  workout.perfs.forEach((perf) => {
    if (perf.perfType === "CLASSIC") {
      out.push(perf.exo);
    }
    if (perf.perfType === "CIRCUIT") {
      perf.exos.forEach((exo) => {
        out.push(exo);
      });
    }
  });
  return out;
};
