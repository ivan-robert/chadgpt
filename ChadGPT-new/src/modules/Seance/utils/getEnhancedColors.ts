import { ThemeInterface } from "#theme/emotion";

export const getEnhancedColors = (
  colors: ThemeInterface["colors"],
  status: "validated" | "disabled" | "neutral"
): ThemeInterface["colors"] => {
  if (status === "validated") {
    return {
      ...colors,
      CTA500: colors.statusSuccess,
      CTA300: colors.statusSuccess,
      text500: colors.statusSuccess,
      text300: colors.statusSuccess,
    };
  }
  if (status === "disabled") {
    return {
      ...colors,
      CTA500: colors.grey200,
      text500: colors.grey200,
      CTA300: colors.grey200,
    };
  }

  return colors;
};
