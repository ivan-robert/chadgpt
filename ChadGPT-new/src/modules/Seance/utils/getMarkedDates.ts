import { Workout } from "#modules/Seance/domain/seance.types";
import { MarkedDates } from "react-native-calendars/src/types";
import { DateTime } from "luxon";
import { Theme } from "@emotion/react";

const getDayDifference = (beforeDate: DateTime, afterDate: DateTime) => {
  const diff = afterDate.startOf("day").diff(beforeDate.startOf("day"), "days");
  return diff.days;
};

export const getMarkedDates = (
  seances: Workout[],
  selected: string,
  theme: Theme
) => {
  const markedDays = seances.reduce((result, seance, index) => {
    const formattedDate = DateTime.fromISO(seance.startDate).toFormat(
      "yyyy-MM-dd"
    );

    const isStartingDay =
      index === seances.length - 1 ||
      !seances.some((s) => {
        return (
          getDayDifference(
            DateTime.fromISO(s.startDate),
            DateTime.fromISO(seance.endDate)
          ) === 1
        );
      });

    const isEndingDay =
      index === 0 ||
      !seances.some(
        (s) =>
          getDayDifference(
            DateTime.fromISO(seance.startDate),
            DateTime.fromISO(s.endDate)
          ) === 1
      );

    result[formattedDate] = {
      marked: true,
      startingDay: isStartingDay,
      endingDay: isEndingDay,
      color: theme.colors.secondary500,
      dotColor: "transparent",
      textColor: theme.colors.text500,
    };

    return result;
  }, {} as MarkedDates);

  markedDays[selected] = {
    ...markedDays[selected],
    startingDay:
      markedDays[selected]?.startingDay !== undefined
        ? markedDays[selected]?.startingDay
        : true,
    endingDay:
      markedDays[selected]?.endingDay !== undefined
        ? markedDays[selected]?.endingDay
        : true,
    selected: true,
    disableTouchEvent: false,
    color: theme.colors.CTA500,
    textColor: theme.colors.white,
  };

  return markedDays;
};
