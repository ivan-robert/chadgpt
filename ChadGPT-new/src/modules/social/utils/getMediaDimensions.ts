import { Media } from "#modules/social/domain/media.types";

type Dimensions = {
  height: number;
  width: number;
};

type ResizeOptions = {
  maxWidth: number;
  maxHeight: number;
};

export const getMediaDimensions = (
  media: Media,
  options: ResizeOptions
): Dimensions => {
  const { height, width } = media;
  const heightFactor = height / options.maxHeight;
  const widthFactor = width / options.maxWidth;
  const factor = Math.max(heightFactor, widthFactor);
  return {
    height: Math.round(height / factor),
    width: Math.round(width / factor),
  };
};
