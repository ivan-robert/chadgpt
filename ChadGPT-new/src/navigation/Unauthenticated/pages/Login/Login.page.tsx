import styled from "@emotion/native";

import { NativeStackScreenProps } from "@react-navigation/native-stack";

import { Spacer } from "#shared/view/components/Spacer";
import { ChadView } from "#modules/auth/view/ChadView";
import { UnauthenticatedNavigatorStackParamList } from "#navigation/Unauthenticated/unauthenticatedNavigator.types";
import { getEnvironment } from "#app/environment";
import { Button } from "#shared/view/components/Button/Button";
import { LoginForm } from "#modules/auth/view/Loginform";
import { PrivacyPolicyText } from "#modules/auth/view/PrivacyPolicy";
import { ChangeLanguageWrapper } from "#modules/Profile/Params/Language/ChangeLanguage.wrapper";

export const LoginPage = ({
  navigation,
}: NativeStackScreenProps<UnauthenticatedNavigatorStackParamList>) => {
  return (
    <>
      <ChadView>
        <ChangeLanguageWrapper>
          <Container>
            <FormContainer>
              <Spacer.Vertical gap={32} />
              <LoginForm
                onCreateAccountPress={() =>
                  navigation.navigate("CreateAccount")
                }
              />
              <Spacer.Vertical gap={32} />
              <PrivacyPolicyText />
            </FormContainer>
          </Container>
        </ChangeLanguageWrapper>
      </ChadView>
      {getEnvironment() === "DEVELOPMENT" && (
        <CheatcodesContainer>
          <Button.Primary
            onPress={() => {
              console.log("not iomplemented chacal");
            }}
            text="log in as Ivan"
          />
        </CheatcodesContainer>
      )}
    </>
  );
};

const FormContainer = styled.View({
  flex: 1,
  justifyContent: "center",
});

const Container = styled.View({
  flex: 1,
  paddingHorizontal: 16,
  paddingVertical: 32,
});

const CheatcodesContainer = styled.View({
  position: "absolute",
  bottom: 100,
  zIndex: 100,
});
