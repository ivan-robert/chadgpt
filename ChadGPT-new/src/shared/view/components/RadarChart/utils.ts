const degToRadians = (degree: number): number => (degree * Math.PI) / 180;

export type Point = [number, number];

export const edgePointCalculatorFor =
  (center: Point, radius: number) =>
  (degree: number, radiusPercentage = 100): Point => {
    const degreeInRadians = degToRadians(degree);
    return [
      center[0] + (Math.cos(degreeInRadians) * radius * radiusPercentage) / 100,
      center[1] + (Math.sin(degreeInRadians) * radius * radiusPercentage) / 100,
    ];
  };
