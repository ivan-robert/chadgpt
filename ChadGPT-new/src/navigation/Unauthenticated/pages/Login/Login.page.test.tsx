import { screen } from "@testing-library/react-native";
import { renderWithProviders } from "../../../../../jest/utils/UtilFunctions";
import { LoginPage } from "./Login.page";
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { UnauthenticatedNavigatorStackParamList } from "../../unauthenticatedNavigator.types";

const navigateMock = jest.fn();
const LoginPageProps = {
  navigation: {
    navigate: navigateMock,
  },
} as unknown as NativeStackScreenProps<UnauthenticatedNavigatorStackParamList>;

describe("Login", () => {
  it("renders correctly", async () => {
    renderWithProviders(<LoginPage {...LoginPageProps} />);
    expect(await screen.findByPlaceholderText("authentication.password"))
      .toBeOnTheScreen;
    expect(screen).toMatchSnapshot();
  });
});
