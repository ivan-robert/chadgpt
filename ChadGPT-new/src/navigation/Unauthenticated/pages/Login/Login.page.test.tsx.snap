// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "height": 1334,
      "position": "absolute",
      "width": 750,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../assets/chad.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": 1334,
          "width": 750,
        },
        undefined,
      ]
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.8)",
        "flex": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "borderColor": "#FFFFFF",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "margin": 8,
          "paddingHorizontal": 8,
          "paddingVertical": 4,
          "position": "absolute",
          "zIndex": 1,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://flagcdn.com/16x12/us.png",
          }
        }
        style={
          {
            "height": 12,
            "width": 16,
          }
        }
      />
      <View
        gap={4}
        style={
          [
            {
              "width": 4,
            },
            undefined,
          ]
        }
      />
      <Text
        accessibilityRole="header"
        style={
          [
            {
              "color": "#FFFFFF",
            },
            {
              "textDecorationLine": "none",
            },
            {
              "flexShrink": 1,
              "fontFamily": "Montserrat-Regular",
              "fontSize": 14,
              "lineHeight": 16.8,
            },
          ]
        }
      >
        profile:settings.changeLanguage
      </Text>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 32,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          gap={32}
          style={
            [
              {
                "height": 32,
              },
              undefined,
            ]
          }
        />
        <Text
          accessibilityRole="header"
          style={
            [
              {
                "color": "#FFFFFF",
              },
              {
                "textDecorationLine": "none",
              },
              {
                "fontFamily": "Montserrat-Bold",
                "fontSize": 40,
                "lineHeight": 48,
              },
            ]
          }
        >
          authentication.login
        </Text>
        <View
          gap={4}
          style={
            [
              {
                "height": 4,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            [
              {
                "gap": 4,
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              isDisabled={false}
              isError={false}
              isFocused={false}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#E1E1E1",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "paddingHorizontal": 8,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "gap": 8,
                      "height": 30,
                      "justifyContent": "center",
                      "width": 30,
                    },
                    undefined,
                  ]
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={30}
                  vbWidth={30}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M24.6593 25.5589C24.0895 23.9641 22.8341 22.5549 21.0876 21.5499C19.3412 20.5448 17.2013 20 15 20C12.7987 20 10.6588 20.5448 8.91239 21.5498C7.16594 22.5549 5.91049 23.9641 5.34074 25.5589"
                      fill={null}
                      propList={
                        [
                          "fill",
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      strokeWidth="2"
                    />
                    <RNSVGCircle
                      cx="15"
                      cy="10"
                      fill={null}
                      propList={
                        [
                          "fill",
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      r="5"
                      stroke={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <TextInput
                autoCapitalize="none"
                autoComplete="email"
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="authentication.email"
                placeholderTextColor="#C1C1C1"
                style={
                  [
                    {
                      "color": "#000000",
                      "flex": 1,
                      "fontSize": 18,
                      "paddingHorizontal": 8,
                      "paddingVertical": 8,
                    },
                    undefined,
                  ]
                }
                textContentType="emailAddress"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          gap={8}
          style={
            [
              {
                "height": 8,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            [
              {
                "gap": 4,
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              isDisabled={false}
              isError={false}
              isFocused={false}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#E1E1E1",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "paddingHorizontal": 8,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "gap": 8,
                      "height": 30,
                      "justifyContent": "center",
                      "width": 30,
                    },
                    undefined,
                  ]
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  fill="none"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={30}
                  vbWidth={30}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M5 15.25C5 13.3644 5 12.4216 5.58579 11.8358C6.17157 11.25 7.11438 11.25 9 11.25H21C22.8856 11.25 23.8284 11.25 24.4142 11.8358C25 12.4216 25 13.3644 25 15.25V20.25C25 23.0784 25 24.4926 24.1213 25.3713C23.2426 26.25 21.8284 26.25 19 26.25H11C8.17157 26.25 6.75736 26.25 5.87868 25.3713C5 24.4926 5 23.0784 5 20.25V15.25Z"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4281546303,
                          "type": 0,
                        }
                      }
                      strokeWidth="2"
                    />
                    <RNSVGPath
                      d="M20 10V8.75C20 5.98858 17.7614 3.75 15 3.75V3.75C12.2386 3.75 10 5.98858 10 8.75V10"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4281546303,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeWidth="2"
                    />
                    <RNSVGCircle
                      cx="15"
                      cy="18.75"
                      fill={
                        {
                          "payload": 4281546303,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                      r="2.5"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <TextInput
                autoCapitalize="none"
                autoComplete="current-password"
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="authentication.password"
                placeholderTextColor="#C1C1C1"
                secureTextEntry={true}
                style={
                  [
                    {
                      "color": "#000000",
                      "flex": 1,
                      "fontSize": 18,
                      "paddingHorizontal": 8,
                      "paddingVertical": 8,
                    },
                    undefined,
                  ]
                }
                textContentType="password"
                value=""
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "gap": 8,
                    "height": 30,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 30,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  fill="none"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGCircle
                      cx="12"
                      cy="12"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      r="3"
                      stroke={
                        {
                          "payload": 4281546303,
                          "type": 0,
                        }
                      }
                      strokeWidth={2}
                    />
                    <RNSVGPath
                      d="M20.188 10.9343C20.5762 11.4056 20.7703 11.6412 20.7703 12C20.7703 12.3588 20.5762 12.5944 20.188 13.0657C18.7679 14.7899 15.6357 18 12 18C8.36427 18 5.23206 14.7899 3.81197 13.0657C3.42381 12.5944 3.22973 12.3588 3.22973 12C3.22973 11.6412 3.42381 11.4056 3.81197 10.9343C5.23206 9.21014 8.36427 6 12 6C15.6357 6 18.7679 9.21014 20.188 10.9343Z"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4281546303,
                          "type": 0,
                        }
                      }
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          gap={4}
          style={
            [
              {
                "height": 4,
              },
              undefined,
            ]
          }
        />
        <View
          gap={16}
          style={
            [
              {
                "height": 16,
              },
              undefined,
            ]
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#141616",
              "borderColor": "#FF7C33",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
              "minHeight": 48,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "gap": 16,
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {},
                  undefined,
                ]
              }
            >
              <Text
                accessibilityRole="header"
                style={
                  [
                    {
                      "color": "#FFFFFF",
                    },
                    {
                      "textDecorationLine": "none",
                    },
                    {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 18,
                      "lineHeight": 21.599999999999998,
                    },
                  ]
                }
              >
                authentication.performLogin
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "aspectRatio": 1,
                    "height": "100%",
                  },
                  undefined,
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                color="#FFFFFF"
                fill="#FFFFFF"
                focusable={false}
                id="arrow"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                tintColor="#FFFFFF"
                vbHeight={15}
                vbWidth={15}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4294967295,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M8.29289 2.29289C8.68342 1.90237 9.31658 1.90237 9.70711 2.29289L14.2071 6.79289C14.5976 7.18342 14.5976 7.81658 14.2071 8.20711L9.70711 12.7071C9.31658 13.0976 8.68342 13.0976 8.29289 12.7071C7.90237 12.3166 7.90237 11.6834 8.29289 11.2929L11 8.5H1.5C0.947715 8.5 0.5 8.05228 0.5 7.5C0.5 6.94772 0.947715 6.5 1.5 6.5H11L8.29289 3.70711C7.90237 3.31658 7.90237 2.68342 8.29289 2.29289Z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          gap={8}
          style={
            [
              {
                "height": 8,
              },
              undefined,
            ]
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
              "minHeight": 48,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "gap": 16,
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {},
                  undefined,
                ]
              }
            >
              <Text
                accessibilityRole="header"
                style={
                  [
                    {
                      "color": "#FFFFFF",
                    },
                    {
                      "textDecorationLine": "none",
                    },
                    {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 18,
                      "lineHeight": 21.599999999999998,
                      "textDecorationLine": "underline",
                    },
                  ]
                }
              >
                authentication.createAccount
              </Text>
            </View>
          </View>
        </View>
        <View
          gap={8}
          style={
            [
              {
                "height": 8,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <Text
            accessibilityRole="header"
            style={
              [
                {
                  "color": "#FFFFFF",
                },
                {
                  "textDecorationLine": "none",
                },
                {
                  "flexShrink": 1,
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16,
                  "lineHeight": 19.2,
                },
              ]
            }
          >
            authentication.or
          </Text>
          <View
            gap={8}
            style={
              [
                {
                  "height": 8,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          gap={32}
          style={
            [
              {
                "height": 32,
              },
              undefined,
            ]
          }
        />
        <Text
          accessibilityRole="header"
          style={
            [
              {
                "color": "#C1C1C1",
              },
              {
                "textDecorationLine": "none",
              },
              {
                "flexShrink": 1,
                "fontFamily": "Montserrat-Regular",
                "fontSize": 14,
                "lineHeight": 16.8,
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
